docs/auth/oauth2-proxy/provider.md

```diff
@@ -0,0 +1,72 @@
+---
+id: provider
+title: OAuth 2 Proxy Provider
+sidebar_label: OAuth 2 Custom Proxy
+description: Adding OAuth2Proxy as an authentication provider in Backstage
+---
+
+The Backstage `core-plugin-api` package comes with an Oauth2Proxy authentication
+provider that can authenticate users by using a
+[oauth2-proxy](https://github.com/oauth2-proxy/oauth2-proxy) in front of an
+actual Backstage instance. This enables to reuse existing authentications within
+a cluster. In general the `oauth2-proxy` supports all OpenID Connect providers,
+for more details check this
+[list of supported providers](https://oauth2-proxy.github.io/oauth2-proxy/docs/configuration/oauth_provider).
+
+## Configuration
+
+The provider configuration can be added to your `app-config.yaml` under the root
+`auth` configuration:
+
+```yaml
+auth:
+  environment: development
+  providers:
+    oauth2proxy: {}
+```
+
+Right now no configuration options are supported. To make use of the provider,
+make sure that the `X-OAUTH2-PROXY-ID-TOKEN` header is set correctly from your
+`oauth2-proxy` instance.
+
+## Adding the provider to the Backstage backend
+
+When using `oauth2proxy` auth you can configure it as described
+[here](https://backstage.io/docs/auth/identity-resolver).
+
+- use the following code below to introduce changes to
+  `packages/backend/plugin/auth.ts`:
+
+```ts
+  providerFactories: {
+    oauth2proxy: createOauth2ProxyProvider<{
+      id: string;
+      email: string;
+    }>({
+      authHandler: async input => {
+        // returns `AuthHandlerResult`
+        return {
+          profile: {
+            email: 'some-email@email.com',
+          },
+        };
+      },
+      signIn: {
+        resolver: async (signInInfo, ctx) => {
+          // returns `BackstageSignInResult`
+          return {
+            id: 'some-id',
+            idToken: 'some-token',
+            token: 'some-token',
```

Comment:
`token` is actually the only thing that's not deprecated in here - might be useful to make this function example more like

```ts
    const token = await ctx.tokenIssuer.issueToken({
      claims: { sub: 'user:default/user-name', ent: [`group:default/optional-user-group`] },
    });
    return { token };
```

or so

Reply:
Yeah, good point. Changed it, but kept the `id` first of all, as it is deprecated, but still required. Will remove it when it's not necessary anymore.